tests/test_isr.py

```diff
@@ -0,0 +1,140 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import os
+import unittest
+import warnings
+
+import lsst.afw.image as afwImage
+import lsst.utils.tests
+
+from lsst.obs.cfht import MegaPrime
+from lsst.obs.cfht.cfhtIsrTask import CfhtIsrTask
+
+
+class CfhtIsrTestCase(lsst.utils.tests.TestCase):
+    """Test for the CFHT IsrTask wrapper."""
+
+    def setUp(self):
+        testDataPackage = "testdata_cfht"
+        try:
+            testDataDirectory = lsst.utils.getPackageDir(testDataPackage)
+        except LookupError as e:
+            warnings.warn(e.args[0])
+            raise unittest.SkipTest("Skipping tests as testdata_cfht is not setup")
+
+        # We'll need a detector, so get the first one.
+        self.camera = MegaPrime().getCamera()
+        detector = self.camera[0]
+
+        # Get the override.
+        self.configPath = os.path.join(lsst.utils.getPackageDir("obs_lsst"), "config",
+                                       "isr.py")
+        # Read the image data.
+        imageLocation = os.path.join(testDataDirectory, "DATA/raw/08BL05/w2.+2+2/2008-11-01/i2",
+                                     "1038843o.fits.fz")
+        self.exposure = self.imageHandler(imageLocation, detector)
+
+    def test_processing(self):
+        # Process image and confirm it doesn't fail.
+
+        config = CfhtIsrTask.ConfigClass()
+        # copy = CfhtIsrTask.ConfigClass()
+        # # Why does load not load the config?  ;_;
+        # copy.load(self.configPath)
+        # config.compare(copy)
+
+        # Manually fix config.
+        config.doDark = False
+        config.doDefect = False
+        config.doFlat = False
+        config.doBias = False
+        config.doFringe = False
+        config.fringeAfterFlat = False
+        config.doWrite = False
+        config.fringe.filters = ["i.MP9701", "i.MP9702", "z.MP9801"]
+        config.fringe.pedestal = True
+        config.fringe.small = 1
+        config.fringe.large = 50
+        config.doAssembleIsrExposures = True
+        config.doSuspect = False
+        config.fluxMag0T1 = {"i.MP9702": 100.0}
+
+        task = CfhtIsrTask(config=config)
+        results = task.run(self.exposure, camera=self.camera)
+        md = results.exposure.getMetadata()
+        amplifiers = list(results.exposure.getDetector().getAmplifiers())
+
+        # Check that saturation level are set correctly.
+        self.assertEqual(md['SATURATE'], 65535)
+
+        # Check that the gain and read noise match what is in the
+        # GAINA and GAINB/RDNOISEA RDNOISEB fields.
+        self.assertEqual(md['GAINA'], amplifiers[0].getGain())
+        self.assertEqual(md['GAINB'], amplifiers[1].getGain())
+        self.assertEqual(md['RDNOISEA'], amplifiers[0].getReadNoise())
+        self.assertEqual(md['RDNOISEB'], amplifiers[1].getReadNoise())
+
+    def imageHandler(self, location, detector=None):
```

Comment:
This should perhaps be a staticmethod, since none of `self`'s attributes are used. Also, this could go either before or after setUp(Class) method, in case we want to add additional tests later.